src/time_utils.c

```diff
@@ -133,9 +133,8 @@ ts_time_value_from_arg(Datum arg, Oid argtype, Oid timetype)
 	else if (argtype != timetype && !can_coerce_type(1, &argtype, &timetype, COERCION_IMPLICIT))
 		ereport(ERROR,
 				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
-				 errmsg("invalid type of time argument"),
-				 errhint("The argument type should be compatible with type \"%s\".",
-						 format_type_be(timetype))));
+				 errmsg("invalid time argument type \"%s\"", format_type_be(argtype)),
+				 errhint("Try casting the argument to \"%s\".", format_type_be(timetype))));
```

Comment:
```suggestion
				 errmsg("invalid time argument type \"%s\"", format_type_be(argtype)),
				 errdetail("The type \"%s\" cannot be implicitly coerced to type \"%s\"", format_type_be(argtype), format_type_be(timetype)),
				 errhint("Try casting the argument to \"%s\".", format_type_be(timetype))));
```

Reply:
"implicitly coerced" seems too implementation-specific. Further, we might change this to `COERCION_ASSIGNMENT`. I would therefore opt to not include these details.

Reply:
My main concern was that it wasn't very clear because only one end of the type coercion was shown. By showing the target type as well it is clear why the coercion does not work, but it is too much details for the simple message but suitable for a detailed message.

Also, since we are using implicit coercion now, I think it is fine to mention this in the detailed message, and change it in the future if we change what coercion we use. It is also a well-know concept for a PG user, so I see no risk of a confusion there.

Reply:
The target type is already shown; the passed in argument type is in the main message, while the target type is in the hint. The detail message you have proposed doesn't add any additional information in this regard. It only displays both types again.